specifyweb/frontend/js_src/lib/components/FormSliders/RecordSet.tsx

```diff
@@ -290,13 +290,25 @@ function RecordSet<SCHEMA extends AnySchema>({
   const [hasDuplicate, handleHasDuplicate, handleDismissDuplicate] =
     useBooleanState();
 
+  const [hasSeveralResourceType, setHasSeveralResourceType] =
+    React.useState(false);
+  const [resourceType, setResourceType] = React.useState<
+    keyof Tables | undefined
+  >(undefined);
+
   async function handleAdd(
     resources: RA<SpecifyResource<SCHEMA>>,
     wasNew: boolean
   ): Promise<void> {
     if (!recordSet.isNew())
       await addIdsToRecordSet(resources.map(({ id }) => id));
     go(totalCount, resources[0].id, undefined, wasNew);
+    if (resourceType === undefined) {
+      setResourceType(resources[0].specifyTable.name);
+    } else if (resourceType !== resources.at(-1)?.specifyTable.name) {
+      setHasSeveralResourceType(true);
+      setResourceType(resources.at(-1)?.specifyTable.name);
+    }
```

Comment:
Do you need `resourceType` for anything apart from setting `hasSeveralResourceType` to true? If not, what do you think about just checking for `resources.length` instead?

Reply:
An idea would have been to do: 
if (new Set(resources.map(resource => resource.name))).size > 1) setHasSeveralResourceType(true) 
But resources is actually an array of one. (the newly added resource) 
We do not have access to the previous resource table name unfortunately without saving it somehow